src/components/MagicCodeInput.tsx

```diff
@@ -264,72 +262,74 @@ function MagicCodeInput(props) {
      * NOTE: when using Android Emulator, this can only be tested using
      * hardware keyboard inputs.
      *
-     * @param {Object} event
+     * @param event
      */
-    const onKeyPress = ({nativeEvent: {key: keyValue}}) => {
+    const onKeyPress = (event: Partial<NativeSyntheticEvent<TextInputKeyPressEventData>>) => {
+        const keyValue = event?.nativeEvent?.key;
         if (keyValue === 'Backspace' || keyValue === '<') {
-            let numbers = decomposeString(props.value, props.maxLength);
+            let numbers = decomposeString(value, maxLength);
 
             // If keyboard is disabled and no input is focused we need to remove
             // the last entered digit and focus on the correct input
-            if (props.isDisableKeyboard && focusedIndex === undefined) {
+            if (isDisableKeyboard && focusedIndex === undefined) {
                 const indexBeforeLastEditIndex = editIndex === 0 ? editIndex : editIndex - 1;
 
                 const indexToFocus = numbers[editIndex] === CONST.MAGIC_CODE_EMPTY_CHAR ? indexBeforeLastEditIndex : editIndex;
-                inputRefs.current[indexToFocus].focus();
-                props.onChangeText(props.value.substring(0, indexToFocus));
+                const formElement = inputRefs.current as HTMLFormElement | null;
+                (formElement?.[indexToFocus] as HTMLInputElement).focus();
+                onChangeTextProp(value.substring(0, indexToFocus));
 
                 return;
             }
 
             // If the currently focused index already has a value, it will delete
             // that value but maintain the focus on the same input.
-            if (numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
+            if (focusedIndex && numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
```

Comment:
It looks like it changes the logic if `focusedIndex` is 0 and can cause problems